passes/hierarchy/hierarchy.cc

```diff
@@ -376,6 +376,65 @@ RTLIL::Module *get_module(RTLIL::Design                  &design,
 	return nullptr;
 }
 
+// Try to read an IdString as a numbered connection name ("$123" or similar),
+// writing the result to dst. If the string isn't of the right format, ignore
+// dst and return false.
+bool read_id_num(RTLIL::IdString str, int *dst)
+{
+	log_assert(dst);
```

Comment:
Would you consider taking `dst` by reference here instead?

Reply:
There's a Google (I think...) style that passes "out parameters" by pointer rather than reference, to ensure that they are more obvious at the call site. The idea is that `ret = read_id_num(name, val)` doesn't obviously modify `val`, but `ret = read_id_num(name, &val)` looks more like it might.

I don't really have strong feelings about it, but I *think* this is slightly clearer. Happy to switch it around if you think that would be better.